Common/TableProducer/trackPropagation.cxx

```diff
@@ -179,45 +192,65 @@ struct TrackPropagation {
-              // trackTunedTracks->Fill(1);
+              q2OverPtNew = mTrackParCov.getQ2Pt();
             }
           }
         }
+        bool isPropagationOK = true;
+
         if (track.has_collision()) {
           auto const& collision = track.collision();
           if constexpr (fillCovMat) {
             mVtx.setPos({collision.posX(), collision.posY(), collision.posZ()});
             mVtx.setCov(collision.covXX(), collision.covXY(), collision.covYY(), collision.covXZ(), collision.covYZ(), collision.covZZ());
-            o2::base::Propagator::Instance()->propagateToDCABxByBz(mVtx, mTrackParCov, 2.f, matCorr, &mDcaInfoCov);
+            isPropagationOK = o2::base::Propagator::Instance()->propagateToDCABxByBz(mVtx, mTrackParCov, 2.f, matCorr, &mDcaInfoCov);
           } else {
-            o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, mTrackPar, 2.f, matCorr, &mDcaInfo);
+            isPropagationOK = o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, mTrackPar, 2.f, matCorr, &mDcaInfo);
           }
         } else {
           if constexpr (fillCovMat) {
             mVtx.setPos({mMeanVtx->getX(), mMeanVtx->getY(), mMeanVtx->getZ()});
             mVtx.setCov(mMeanVtx->getSigmaX() * mMeanVtx->getSigmaX(), 0.0f, mMeanVtx->getSigmaY() * mMeanVtx->getSigmaY(), 0.0f, 0.0f, mMeanVtx->getSigmaZ() * mMeanVtx->getSigmaZ());
-            o2::base::Propagator::Instance()->propagateToDCABxByBz(mVtx, mTrackParCov, 2.f, matCorr, &mDcaInfoCov);
+            isPropagationOK = o2::base::Propagator::Instance()->propagateToDCABxByBz(mVtx, mTrackParCov, 2.f, matCorr, &mDcaInfoCov);
           } else {
-            o2::base::Propagator::Instance()->propagateToDCABxByBz({mMeanVtx->getX(), mMeanVtx->getY(), mMeanVtx->getZ()}, mTrackPar, 2.f, matCorr, &mDcaInfo);
+            isPropagationOK = o2::base::Propagator::Instance()->propagateToDCABxByBz({mMeanVtx->getX(), mMeanVtx->getY(), mMeanVtx->getZ()}, mTrackPar, 2.f, matCorr, &mDcaInfo);
           }
         }
-        trackType = aod::track::Track;
+        if (isPropagationOK) {
+          trackType = aod::track::Track;
+        }
+
+        // filling some QA histograms for track tuner test purpose
+        if constexpr (isMc && fillCovMat) { /// track tuner ok only if cov. matrix is used
+          if (track.has_mcParticle() && isPropagationOK) {
+            auto mcParticle1 = track.mcParticle();
+            // && abs(mcParticle1.pdgCode())==211
+            if (mcParticle1.isPhysicalPrimary() && abs(mcParticle1.pdgCode()) == 211) {
+              registry.fill(HIST("hDCAxyVsPtRec"), mDcaInfoCov.getY(), mTrackParCov.getPt());
+              registry.fill(HIST("hDCAxyVsPtMC"), mDcaInfoCov.getY(), mcParticle1.pt());
+            }
+          }
+        }
+      }
+      // Filling modified Q/Pt values by track tuner in track tuner table
```

Comment:
```suggestion
      // Filling modified Q/Pt values at IU/production point by track tuner in track tuner table
```suggestion
      // Filling modified Q/Pt values by track tuner in track tuner table
```
```